handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/filterext/WarmUpDiscoveryFilter.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.loadbalance.filterext;
+
+import com.netflix.config.DynamicPropertyFactory;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.servicecomb.core.Invocation;
+import org.apache.servicecomb.loadbalance.ServerListFilterExt;
+import org.apache.servicecomb.loadbalance.ServiceCombServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+public class WarmUpDiscoveryFilter implements ServerListFilterExt {
+  private static final Logger LOGGER = LoggerFactory.getLogger(WarmUpDiscoveryFilter.class);
+
+  private static final int INSTANCE_WEIGHT = 100;
+
+  // Default time for warm up, the unit is second
+  private static final String DEFAULT_WARM_UP_TIME = "30";
+
+  private static final String WARM_TIME_KEY = "warmupTime";
+
+  private static final String WARM_CURVE_KEY = "warmupCurve";
+
+  // Preheat calculates curve value
+  private static final String DEFAULT_WARM_UP_CURVE = "2";
+
+  private final Random random = new Random();
+
+  @Override
+  public int getOrder() {
+    return ORDER_WARM_UP;
+  }
+
+  @Override
+  public boolean enabled() {
+    return DynamicPropertyFactory.getInstance()
+        .getBooleanProperty(WARM_UP_FILTER_ENABLED, true)
+        .get();
+  }
+
+  @Override
+  public List<ServiceCombServer> getFilteredListOfServers(List<ServiceCombServer> servers,
+      Invocation invocation) {
+    if (servers.size() <= 1) {
+      return servers;
+    }
+    boolean isAllInstanceWarmUp = true;
+    int[] weights = new int[servers.size()];
+    int totalWeight = 0;
+    int index = 0;
+    for (ServiceCombServer server : servers) {
+      Map<String, String> properties = server.getInstance().getProperties();
+      String registerTimeStr = StringUtils.isBlank(server.getInstance().getTimestamp()) ? "0" :
+              server.getInstance().getTimestamp();
```

Comment:
Not all Discovery implementation has timestamp and service center is server time not application time.

Reply:
change to set register time by instance property